File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.java
import java.util.Set;
import java.util.HashSet;

class Solution {
    public int solution(int[] nums) {
        int answer = 0;
        Set<Integer> set = new HashSet<>();
        for(int i = 0 ; i < nums.length; i ++ ) set.add(nums[i]);
        answer = set.size() > nums.length/2 ? nums.length/2 : set.size();
        return answer;
    }
}

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.cpp
#include <vector>
#include <map>
using namespace std;

int solution(vector<int> nums)
{
    int answer = 0;
    
    map<int, int> temp;
    for(int i = 0 ; i < nums.size() ; i ++ ){
        temp[nums[i]]++;
    }
    
    if(temp.size() >= (nums.size()/2)){
        answer = (nums.size())/2;
    }
    else{
        answer = temp.size();
    }
    return answer;
}